paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaymentAdminController.java

```diff
@@ -116,11 +119,16 @@ public ResponseEntity<Void> orderCreatedEventCallbackPaytrail(@RequestBody Order
         try {
             log.debug("payment-api received ORDER_CREATED event for orderId: " + message.getOrderId());
 
+
             Payment existingPayment = onlinePaymentService.getPaymentForOrder(message.getOrderId());
             if (existingPayment != null && PaymentStatus.PAID_ONLINE.equals(existingPayment.getStatus())) {
                 log.warn("paid payment exists, not creating new payment for orderId: " + message.getOrderId());
             } else {
-                if (Boolean.TRUE.equals(message.getIsSubscriptionRenewalOrder()) && message.isCardDefined()) {
+                if ( message.getEndDate().toLocalDate().isBefore(LocalDate.now(ZoneId.of("Europe/Helsinki"))) ){
```

Comment:
It is UTC-0 and that causes the problem. It is end of day in utc 0 so I cannot directly compare to it since JAVA understands timezones and different timezones can be compared directly otherwise I would be checking against 23:59:59 in Greenwitch which would be 2 or 3 hours off. Java would see that 0:00:00 in Helsinki is same as 22:00:00 in utc so direct comparison would still try to pay at 2AM or 3AM Finnish time (because that would be midnight in UTC). LocalDate does not have timezone so I can take date from endDate and get date now in Helsinki and see if end date has passed.